File: python/multiple_alarms.py
import alarm
from alarm import Alarm
from interfaces import Alarm_Behavior


class Multiple_Alarms(Alarm_Behavior):
    def __init__(self):
        self._alarms = []

    def add_alarm(self, alarm):
        self._alarms.append(alarm)

    def get_alarms(self, from_time=None, to_time=None):
        return [alarm for alarm in self._alarms
                if (from_time is None or alarm.alarm_time >= from_time) and
                (to_time is None or alarm.alarm_time <= to_time)]

    def remove_alarm(self, alarm):
        raise NotImplementedError

    def find_alarm_by_name(self, alarm_name):
        raise NotImplementedError

File: java/src/com/infoback/MultipleAlarms.java
package com.infoback;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

public class MultipleAlarms implements IAlarmStrategy {
  private List<Alarm> _alarms = new ArrayList<Alarm>();

  @Override
  public void addAlarm(Alarm alarm) {
    _alarms.add(alarm);
  }

  @Override
  public Iterable<Alarm> getAlarms(LocalDateTime fromTime, LocalDateTime toTime) {
    return _alarms.stream()
      .filter(alarm -> (fromTime == null || alarm.getAlarmTime().isAfter(fromTime)) &&
                       (toTime == null || alarm.getAlarmTime().isBefore(toTime)))
      .collect(Collectors.toList());
  }

  @Override
  public void removeAlarm(Alarm alarm) {
    throw new UnsupportedOperationException();
  }

  @Override
  public Alarm findAlarmByName(String alarmName) {
    throw new UnsupportedOperationException();
  }
}
